src/Chainweb/Pact/PactService.hs

```diff
@@ -949,16 +949,17 @@ toHashCommandResult = over (P.crLogs . _Just) $ P.pactHash . encodeToByteString
 
 transactionsFromPayload :: PayloadData -> IO (Vector ChainwebTransaction)
 transactionsFromPayload plData = do
-    let !transSeq = _payloadDataTransactions plData
-    let !transList = toList transSeq
-    let !bytes = _transactionBytes <$!> transList
-    let !eithers = toCWTransaction <$!> bytes
-    -- Note: if any transactions fail to convert, the final validation hash
-    -- will fail to match the one computed during newBlock
-    let theRights = rights eithers
-    return $! V.fromList theRights
+    unless (null theLefts) $ do
+        throwM $ TransactionDecodeFailure $ "Failed to decode pact transactions: "
```

Comment:
I'm not sure that throwing is the right thing here.  It might be, but we should run it by @gregorycollins and @slpopejoy

Reply:
We reach this code only after either the mempool parsed the tx or we got the block from another node. So, if parsing fails, it's almost certainly due to the latter, in which case we reject the block. Throwing an exception during pact validation is a way to reject the block.

(Currently, we don't blacklist the sender, so we may try to validate the same block again latter, but that problem is a broader issue and isn't specific to this failure case)